// plugin-welcome/src/api/models/EntSystemmemberEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntPrescription,
    EntPrescriptionFromJSON,
    EntPrescriptionFromJSONTyped,
    EntPrescriptionToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntSystemmemberEdges
 */
export interface EntSystemmemberEdges {
    /**
     * Prescriptions holds the value of the prescriptions edge.
     * @type {Array<EntPrescription>}
     * @memberof EntSystemmemberEdges
     */
    prescriptions?: Array<EntPrescription>;
}

export function EntSystemmemberEdgesFromJSON(json: any): EntSystemmemberEdges {
    return EntSystemmemberEdgesFromJSONTyped(json, false);
}

export function EntSystemmemberEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntSystemmemberEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'prescriptions': !exists(json, 'prescriptions') ? undefined : ((json['prescriptions'] as Array<any>).map(EntPrescriptionFromJSON)),
    };
}

export function EntSystemmemberEdgesToJSON(value?: EntSystemmemberEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'prescriptions': value.prescriptions === undefined ? undefined : ((value.prescriptions as Array<any>).map(EntPrescriptionToJSON)),
    };
}
